go/vt/sysvars/sysvars.go

```diff
@@ -39,8 +39,18 @@ type SystemVariable struct {
 	Name string
 
 	SupportSetVar bool
+
+	Case StorageCase
 }
 
+type StorageCase int
+
+const (
+	SCSame StorageCase = iota
```

Comment:
nitpick: Couldn't we skip the `SC` and just use `sysvars.AlwaysUpper`,`sysvars.AlwaysLower` and `sysvars.FromUser`? really not important, if you want to merge, go ahead :)